### ISSUE TITLE
mso_schema_template_bd - error Invalid Virtual MAC address during BD creation

### ISSUE THREAD

**Issue by russomichele** [CiscoDevNet/terraform-provider-mso#163]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v1.3.5

### MSO version

* 3.1.1n

### APIC version and APIC Platform for Site Level Resources

* V 5.2(5e) on-prem.

### Affected Resource(s)

* mso_schema_template_bd

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
resource "mso_schema_template_bd" "[ OMIT ]" {
	schema_id              = [ OMIT ]
	template_name          = [ OMIT ]
	name                   = [ OMIT ]
	display_name           = [ OMIT ]
	vrf_name               = [ OMIT ]
	layer2_unknown_unicast = [ OMIT ]
	vrf_schema_id          = [ OMIT ]
	vrf_template_name      = [ OMIT ]
}

```

### Debug Output

Error: "Bad Request: Patch Failed, Received: Invalid Virtual MAC address for BD '[ OMIT ]' in template [ OMIT ] exception while trying to update schema"

### Expected Behavior

BD created without vmac

### Actual Behavior
Bad request 

### Steps to Reproduce

1. `terraform apply`

### Important Factoids
Not working since MSO provider 0.5.0.

**Comment by apellini** [comment]
Hi,
have any news on that?

**Comment by niksheridan-ops** [comment]
(note I include ```TF_CLI_ARGS_apply="-parallelism=1"``` as and environment variable)
When using 
```TF_LOG=debug terraform apply```
there is a patch request created as expected - I am unable to paste this in (sorry I cant to this due to business requirements) but the body of the request for the api call includes this particular value - an empty string (as opposed no none/null)  

```
"vmac":""
```
I think this resource appears to be reworked between 0.6 and 0.7 - should this be null or none as opposed to a empty string?  I was a bit worried the API might not accept this an return the error (I have no yet curl'ed the equivalent)

I think in v0.7 this is the line that may be relevant
v0.7x
https://github.com/CiscoDevNet/terraform-provider-mso/blob/94ec11c48ab69fe076617d8cfd4b8d742610a277/mso/resource_mso_schema_template_bd.go#L274

v0.6x cant seem to find equivalent else statement?
https://github.com/CiscoDevNet/terraform-provider-mso/blob/7c7b3c4485530cad039548a083f83bc5fd76322c/mso/resource_mso_schema_template_bd.go#L433

**Comment by niksheridan-ops** [comment]
Kudos to @russomichele  I see your PR https://github.com/CiscoDevNet/terraform-provider-mso/pull/164/files many thanks indeed and it seems we are on a similar wave length

**Comment by apellini** [comment]
We have seen a comment that indicates to have MSO 3.2 but we have 3.1.1n release and at the moment we could not have a plan to upgrade. Could you test with our release and accept the pr?

**Comment by akinross** [comment]
Hi @apellini,

I am not sure that I follow what you mean exactly, which comment indicates MSO 3.2? 

I have ran tests on 3.1(1n) MSO with the latest code of the MSO TF provider from this repository. This will soon be released as a successor of MSO TF provider release 0.7.1. The issue is already fixed in a commit in master, which was done after the MSO TF provider 0.7.1 release, thus accepting the PR and merging it into master would make no difference. 

Kind regards,

Akini

**Comment by apellini** [comment]
Hi,
On line 36 and 37 there is this comment schema template bd go file. I’m happy to know this. Thanks a lot to test it! We’ll try on our lab. If you agree we mantain this issue until test done on our env with 0.7.1 provider release.

**Comment by lhercot** [comment]
@apellini @niksheridan @russomichele Please try the new version of the provider v0.8.1 and let us know if you still have this issue with it.

**Comment by apellini** [comment]
Yes it works for us

**Comment by lhercot** [comment]
@russomichele is it fixed for you too using the new provider v0.8.1?

**Comment by russomichele** [comment]
Hi, yes I can confirm that with the issue is fixed with provider version 0.8.0 and 0.8.1 with MSO 3.1.1n.
Thank you
